### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by JimboJoe** [comment]
!testme

**Comment by JimboJoe** [comment]
We're getting there: the LDAP plugin configuration is fixed, and there's a last remaining problem in the upgrade with the files-keeping mechanism...

**Comment by JimboJoe** [comment]
!testme

**Comment by JimboJoe** [comment]
@ericgaspar I don't understand why the saved `_data` is not a directory in the CI. Installing manually creates a populated `_data` directory, and force-upgrading does work...

**Comment by JimboJoe** [comment]
In my understanding:

- in the old days, the data directories (_data, galleries, uploads) were store in /home/yunohost.app/piwigo and a symbolic link was created in the install dir (/var/www/piwigo)
- Then the upstream app started managing a "data dir" config, and we set it up, not needing the symbolic links any more; the data dirs were still stored in /home/yunohost.app/piwigo

So... in any way, the data directories are kept during the upgrade, so we shouldn't need to preserve the corresponding ones in the install dir (not used) during the upstream source uncompressing (revert from [that change](https://github.com/YunoHost-Apps/piwigo_ynh/pull/129/files#diff-57aeb84da86cb7420dfedd8e49bc644fb799d5413d01927a0417bde753e8922fR28)). And it should also fix the remaining "upgrade from older version" tests

@ericgaspar Does that sound legit...?

File: scripts/upgrade [hunk 1]
```diff
@@ -25,15 +25,10 @@ fi
 #=================================================
 ynh_script_progression --message="Upgrading source files..." --weight=3
 
-ynh_setup_source --dest_dir="$install_dir" --full_replace=1 --keep="local/config/config.inc.php local/config/database.inc.php"
+ynh_setup_source --dest_dir="$install_dir" --full_replace=1 --keep="local/config/config.inc.php local/config/database.inc.php _data galleries upload"
 ynh_setup_source --dest_dir="$install_dir/plugins/Ldap_Login" --source_id=ldap_plugin --full_replace=1
 ynh_setup_source --dest_dir="$install_dir/plugins" --source_id=log_failed_logins_plugin --full_replace=1
```
Comment (ericgaspar) [inline scripts/upgrade:30]: We should remove those `--full_replace` on line 29 and 30 as line 28 already cleans the directory. (and line 30 erase the line 29 plugins file)

**Comment by ericgaspar** [comment]
To be honest, I don't remember exactly what the issue was on the Master branch... something to do with images not showing up due to symlinks not working.
This PR removes the use of `data_dir` in favor of the `install_dir/(_data, galleries, uploads)` location.
That being said, I'm not sure if this is the right strategy and if upgrading won't wipe out existing data...

**Comment by JimboJoe** [comment]
Ah I see the analysis in https://github.com/YunoHost-Apps/piwigo_ynh/issues/128#issuecomment-2156085035
Then, a potentially simpler/less risky solution couldn't be to configure the `data_location` and `upload_dir` settings in the app with a relative path (i.e. `../../..__DATA_DIR__`)...?

**Comment by ericgaspar** [comment]
ah ok. I understand. I commited #141. Along with the migration to 2.1 helpers, the `_data, galleries, uploads` files were created in the `data_dir` location and `$conf['data_location'] = '../../..__DATA_DIR__/_data/';` is added to the config file. Let's see if that does the trick

**Comment by ericgaspar** [comment]
!testme

**Comment by JimboJoe** [comment]
Is that branch ready for larger testing (I have no personal instance using piwigo)...?

**Comment by ericgaspar** [comment]
I haven't installed Piwigo myself, but yes, it's ready for worldwide testing!

**Comment by Marc-Flender** [comment]
Bonjour, 
cette application n'est plus au catalogue Yunohost, malheureusement
Que manque-t-il pour la rajouter?

**Comment by ericgaspar** [comment]
!testme

**Pull request merged**